idaes/models/properties/modular_properties/phase_equil/smooth_VLE_2.py

```diff
@@ -0,0 +1,278 @@
+    """
+
+    @staticmethod
+    def phase_equil(b, phase_pair):
+        # This method is called via StateBlock.build, thus does not need clean-up
+        # try/except statements
+        suffix = "_" + phase_pair[0] + "_" + phase_pair[1]
+
+        # Smooth VLE assumes a liquid and a vapor phase, so validate this
+        (
+            l_phase,
+            v_phase,
+            vl_comps,
+            henry_comps,
+            l_only_comps,
+            v_only_comps,
+        ) = identify_VL_component_list(b, phase_pair)
+
+        if l_phase is None or v_phase is None:
+            raise ConfigurationError(
+                f"{b.params.name} Generic Property Package phase pair {phase_pair[0]}-{phase_pair[1]} "
+                "was set to use Smooth VLE formulation, however this is not a vapor-liquid pair."
+            )
+
+        # Definition of equilibrium temperature for smooth VLE
+        uom = b.params.get_metadata().default_units
+        t_units = uom.TEMPERATURE
+        f_units = uom.AMOUNT / uom.TIME
+
+        s = Var(
+            b.params.phase_list,
+            initialize=0.0,
+            bounds=(0, None),
+            doc="Slack variable for equilibrium temperature",
+            units=pyunits.dimensionless,
+        )
+        b.add_component("s" + suffix, s)
+
+        # Equilibrium temperature
+        def rule_teq(b):
+            if b.params.get_phase(phase_pair[0]).is_vapor_phase():
+                vapor_phase = phase_pair[0]
+                liquid_phase = phase_pair[1]
+            else:
+                vapor_phase = phase_pair[1]
+                liquid_phase = phase_pair[0]
+            return (
+                b._teq[phase_pair]
+                - b.temperature
+                - s[vapor_phase] * t_units
+                + s[liquid_phase] * t_units
+                == 0
+            )
+
+        b.add_component("_tbar_constraint" + suffix, Constraint(rule=rule_teq))
+
+        eps = Param(
+            default=1e-04,
+            mutable=True,
+            doc="Smoothing parameter for complementarities",
+            units=f_units,
+        )
+        b.add_component("eps" + suffix, eps)
+
+        gp = Var(
+            b.params.phase_list,
+            initialize=0.0,
+            bounds=(0, None),
+            doc="Slack variable for cubic second derivative for phase p",
+            units=pyunits.dimensionless,
+        )
+        b.add_component("gp" + suffix, gp)
+
+        gn = Var(
+            b.params.phase_list,
+            initialize=0.0,
+            bounds=(0, None),
+            doc="Slack variable for cubic second derivative for phase p",
+            units=pyunits.dimensionless,
+        )
+        b.add_component("gn" + suffix, gn)
+
+        def rule_temperature_slack_complementarity(b, p):
+            flow_phase = b.flow_mol_phase[p]
+
+            return smooth_min(s[p] * f_units, flow_phase, eps) == 0
```

Comment:
Okay, we're treating `s` as having different units depending on the context. That's not ideal for scaling.

Reply:
Agreed - if you have any ideas on how to handle this better it would be appreciated. For now I have just focused on getting the code running.

Reply:
My instinct here is to just give `s[p]` units of temperature per flowrate, then scale this equation like temperature. However, it needs to be scaled with a "temperature difference" scaling factor, not an "absolute temperature" scaling factor.

Reply:
Based on some testing, scaling of s, g and epsilon depends on the larger of T and flow, thus it is hard to know what to do with these. I am inclined to leave these as unit-less for that reason, and any scaling routine can handle the necessary logic behind the scenes (as I doubt users will care enough to understand what is happening).